### PR TITLE
Add joinLines.

### PR DESCRIPTION
**Pull request by jacereda**

### CODE REVIEW

File: core/carp_pattern.h [hunk 1]
```diff
@@ -87,9 +87,7 @@ String Pattern_internal_classend(PatternMatchState *ms, String p) {
             } while (*p != ']');
             return p + 1;
         }
-        default: {
-            return p;
-        }
+        default: { return p; }
```
Comment (jacereda) [inline core/carp_pattern.h:90]: I always have to discard this diff before committing. My `clang-format` version must be different from the one you're using. I'd suggest getting rid of the braces, maybe both versions will behave the same in that case.

Comment (jacereda) [reply core/carp_pattern.h:90]: My version is 7.1.0 BTW.

Comment (jacereda) [reply core/carp_pattern.h]: Just pushed it without the braces, can someone confirm older versions format it the same?

Comment (hellerve) [reply core/carp_pattern.h]: Actually, my version (10.0.0) does not reformat that line.

Comment (jacereda) [reply core/carp_pattern.h]: Oh, right, I'll update my version then.

**Pull request merged**